src/components/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,145 @@
+import { ComponentProps, forwardRef, createContext, useCallback, useState, useEffect, KeyboardEvent } from 'react';
+import { getCommonProps } from '../../utils';
+import classnames from 'classnames';
+import useEmblaCarousel, { type UseEmblaCarouselType } from 'embla-carousel-react';
+
+type CarouselApi = UseEmblaCarouselType[1];
+
+// to expose more options from the embla carousel API
+// see what is available in https://www.embla-carousel.com/api/options/
+export interface CarouselProps extends ComponentProps<'div'> {
+  /**
+   * Whether the carousel should loop.
+   */
+  loop?: boolean;
+  /**
+   * The index to start the carousel at. Can be used if you want the carousel to be controlled.
+   */
+  startIndex?: number;
+  /**
+   * Function to call when the slide changes.
+   */
+  onSlideChange?: (index: number) => void;
+}
+
+type CarouselContextProps = {
+  carouselRef: ReturnType<typeof useEmblaCarousel>[0];
+  api: ReturnType<typeof useEmblaCarousel>[1];
+  scrollPrev: () => void;
+  scrollNext: () => void;
+  canScrollPrev: boolean;
+  canScrollNext: boolean;
+} & CarouselProps;
+
+export const CarouselContext = createContext<CarouselContextProps | null>(null);
+
+/**
+ * ## Overview
+ *
+ * Carousel component
+ *
+ * Combine with CarouselContent, CarouselItem, and CarouselDots to use
+ *
+ * API Inspired by https://ui.shadcn.com/docs/components/carousel#api
+ *
+ * Built on top of https://github.com/davidjerleke/embla-carousel
+ *
+ * [Figma Link](https://www.figma.com/design/hMu9IWH5N3KamJy8tLFdyV/EASEL-Compendium%3A-Tokens%2C-Components-%26-Patterns?node-id=7068-6007&t=HYwLAd3xkKAOgjpo-1)
+ *
+ * [Storybook Link](https://phillips-seldon.netlify.app/?path=/docs/components-carousel--overview)
+ */
+const Carousel = forwardRef<HTMLDivElement, CarouselProps>(
+  ({ loop = false, startIndex = 0, onSlideChange, className, children, ...props }, ref) => {
+    const { className: baseClassName, ...commonProps } = getCommonProps(props, 'Carousel');
+
+    const [carouselRef, api] = useEmblaCarousel({
+      loop,
+      startIndex,
+    });
+    const [canScrollPrev, setCanScrollPrev] = useState(false);
+    const [canScrollNext, setCanScrollNext] = useState(false);
+
+    const onSelect = useCallback((api: CarouselApi) => {
+      if (!api) {
+        return;
+      }
+
+      setCanScrollPrev(api.canScrollPrev());
+      setCanScrollNext(api.canScrollNext());
+    }, []);
+
+    const scrollPrev = useCallback(() => {
+      api?.scrollPrev();
```

Comment:
the extra `useCallback` might not be needed here as we shouldn't have too many children, we would have to `React.memo` the children to make this work anyway